File: pegasusX/apps/retailer-app-android/app/src/main/java/com/pegasusx/retailer/service/EnterpriseUpdateConfig.kt
package com.pegasusx.retailer.service

import com.pegasusx.retailer.BuildConfig

/**
 * Distribution config for native Android.
 *
 * - **enterprise** flavor: website CDN OTA (`channel=enterprise`, ENABLE_CDN_OTA=true)
 * - **store** flavor: Play Store (`channel=production`, ENABLE_CDN_OTA=false)
 *
 * Build with: `./gradlew :app:assembleStoreRelease` or `assembleEnterpriseRelease`.
 */
object EnterpriseUpdateConfig {
    /** Client-policy channel — production for Play, enterprise for website. */
    val CHANNEL: String
        get() = BuildConfig.DISTRIBUTION_CHANNEL

    /** When false, never download/install APK from CDN; open Play listing instead. */
    val enableCdnOta: Boolean
        get() = BuildConfig.ENABLE_CDN_OTA

    /** Policy role for client-policy API. */
    const val POLICY_ROLE = "RETAILER"

    /**
     * Fallback CDN manifest for enterprise only.
     * Store builds ignore this and use [storeListingUrl].
     */
    const val DEFAULT_MANIFEST_URL =
        "https://storage.googleapis.com/pegasusx-ssmr-app-updates/android/retailer/updater.json"

    /** Play Store listing (store flavor); empty on enterprise flavor. */
    val storeListingUrl: String
        get() = BuildConfig.STORE_LISTING_URL.ifBlank {
            "https://play.google.com/store/apps/details?id=${BuildConfig.APPLICATION_ID}"
        }

    val isStoreBuild: Boolean
        get() = !enableCdnOta
}

File: pegasusX/apps/retailer-app-ios/retailerapp/reatilerapp/Services/EnterpriseUpdateConfig.swift
import Foundation

/// Distribution config for native iOS.
///
/// **Website enterprise (default):** `channel=enterprise`, CDN itms-services OTA.
/// **App Store:** set Info.plist `PXDistributionChannel` = `production`
///   and `PXAppStoreURL` (or `PXAppStoreID`), or compile with `-DSTORE_BUILD`.
///
/// Do not ship App Store binaries with enterprise CDN OTA enabled.
enum EnterpriseUpdateConfig {
    /// Client-policy channel (`production` for App Store, `enterprise` for website).
    static var channel: String {
        if let v = Bundle.main.object(forInfoDictionaryKey: "PXDistributionChannel") as? String,
           !v.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty {
            return v.trimmingCharacters(in: .whitespacesAndNewlines).lowercased()
        }
        #if STORE_BUILD
        return "production"
        #else
        return "enterprise"
        #endif
    }

    /// When false, open App Store listing instead of itms-services CDN install.
    static var enableCdnOta: Bool {
        if let v = Bundle.main.object(forInfoDictionaryKey: "PXEnableCdnOta") as? Bool {
            return v
        }
        if let s = Bundle.main.object(forInfoDictionaryKey: "PXEnableCdnOta") as? String {
            return (s as NSString).boolValue
        }
        #if STORE_BUILD
        return false
        #else
        return channel != "production" && channel != "store"
        #endif
    }

    static let policyRole = "RETAILER"

    static let defaultManifestURL = URL(
        string: "https://storage.googleapis.com/pegasusx-ssmr-app-updates/ios/retailer/updater.json"
    )!

    /// App Store listing URL for production builds.
    static var storeListingURL: URL? {
        if let s = Bundle.main.object(forInfoDictionaryKey: "PXAppStoreURL") as? String,
           let u = URL(string: s), !s.isEmpty {
            return u
        }
        if let id = Bundle.main.object(forInfoDictionaryKey: "PXAppStoreID") as? String {
            let trimmed = id.trimmingCharacters(in: .whitespacesAndNewlines)
                .replacingOccurrences(of: "id", with: "", options: [.anchored])
            if !trimmed.isEmpty {
                return URL(string: "https://apps.apple.com/app/id\(trimmed)")
            }
        }
        return nil
    }

    static var isStoreBuild: Bool { !enableCdnOta }
}
